gprofiler/application_separators.py

```diff
@@ -0,0 +1,193 @@
+            if arg.endswith(".ini"):
+                config_file = arg
+
+        if config_file is None:
+            return None
+
+        if not os.path.isabs(config_file):
+            config_file = os.path.join(process.cwd(), config_file)
+
+        config = configparser.ConfigParser()
+        config.read(resolve_host_path(process, config_file))
+        try:
+            return config.get("uwsgi", "module")
+        except (configparser.NoSectionError, configparser.NoOptionError):
+            pass
+
+        return None
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        wsgi = get_positional_arg(process.cmdline(), "-w")
+        if wsgi is not None:
+            return wsgi
+
+        wsgi = self._find_wsgi_from_config_file(process)
+        if wsgi is not None:
+            return wsgi
+
+        logger.warning("Couldn't find uwsgi wsgi module, both from cmdline and from config file")
+        return None
+
+
+class CeleryApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "celery"
+
+    def is_supported(self, process: Process) -> bool:
+        if process.cmdline()[0] == "celery":
+            return True
+
+        return len(process.cmdline()) >= 3 and ["-m", "celery"] == process.cmdline()[1:]
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        app_name = get_positional_arg(process.cmdline(), "-A") or get_positional_arg(process.cmdline(), "--app")
+        if app_name is None:
+            logger.warning("Couldn't find positional argument -A or --app for application indication")
+            return None
+
+        return app_name
+
+
+class PySparkApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "pyspark"
+
+    def is_supported(self, process: Process) -> bool:
+        # We're looking for pythonXX -m pyspark.daemon
+        return process.cmdline() >= 3 and ["-m", "pyspark.daemon"] == process.cmdline()[1:]
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        return "generic pyspark.daemon"
+
+
+class PythonModuleApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "generic-python"
+
+    def is_supported(self, process: Process) -> bool:
+        if "python" not in process.cmdline()[0]:
+            return False
+
+        return (len(process.cmdline()) > 3 and process.cmdline()[1] == "-m") or (
+            len(process.cmdline()) >= 2 and process.cmdline()[1].endswith(".py")
+        )
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        module_arg = get_positional_arg(process.cmdline(), "-m")
+        if module_arg is not None:
+            return module_arg
+
+        module_filename = process.cmdline()[1][:-3]  # Strip the ".py" (checked before)
+        return append_python_module_to_proc_wd(process, module_filename)
```

Comment:
Running `python3.8 /usr/local/lib/python3.8/dist-packages/my-module.py` from `/home/myself` would've get `".home.myself/usr/local/lib/python3.8/dist-packages/my-module"`. I think `os.path.basename(process.cmdline()[1])[:-3]` would've sufficed here.

Reply:
The idea was to add the cwd to the module name since we can have 2 different applications with the same basename in 2 different dirs.

Reply:
[This comment](https://github.com/Granulate/gprofiler/pull/242#discussion_r774495062)

Reply:
fixed

Reply:
Not pushed yet

Reply:
Okay, it is pushed and fixed `_PythonModuleApplicationIdentifier`, not sure about the others tho. But here it's okay now. Resolving.